### PR TITLE
Generalized `no-goto-without-base` into `no-navigation-without-base`

### LINKED ISSUE #891: Add rule `no-link-without-base`
### Motivation

The motivation is basically similar to #675 - when doing *internal* navigation, you almost always want to prepend the base path. However, unlike `goto`, links can also be used for external navigation. To navigate this complicated issue, I propose to only check relative links (at least by default).

### Description

Add a rule that would trigger on any link without a base path. However, there are quite some edge cases

### Examples

```svelte
<script>
  import { base } from "$app/paths";
  import { base as whatever } from "$app/paths";
  import { goto } from "$app/navigation";
</script>

  <!-- ✓ GOOD -->
<a href={base + "/foo"}>Text</a>
<a href={`${base}/foo`}>Text</a>
<a href={whatever + "/foo"}>Text</a>
<a href={`${whatever}/foo`}>Text</a>
<a href="https://absolute.url">Text</a>

  <!-- ✗ BAD -->
<a href={"/foo"}>Text</a>
<a href={"/foo" + base}>Text</a>
<a href={`foo/${base}`}>Text</a>
</svelte>
```

### Additional comments

Based on https://github.com/sveltejs/eslint-plugin-svelte/pull/679#issuecomment-1975067137

### PR DESCRIPTION
**Pull request by marekdedic**
Closes #891

### CODE REVIEW

File: packages/eslint-plugin-svelte/src/rules/no-navigation-without-base.ts [hunk 1]
```diff
@@ -0,0 +1,331 @@
...
+	const set = new Set<TSESTree.Identifier>();
+	for (const { node } of referenceTracker.iterateEsmReferences({
+		'$app/paths': {
+			[ReferenceTracker.ESM]: true,
+			base: {
+				[ReferenceTracker.READ]: true
+			}
+		}
+	})) {
+		const variable = findVariable(context, (node as TSESTree.ImportSpecifier).local);
```
Comment (ota-meshi) [inline packages/eslint-plugin-svelte/src/rules/no-navigation-without-base.ts:130]: Could you please add a test case to see if the following import works?

```js
	import *  as paths from '$app/paths';
    paths.base;
```

**Review by ota-meshi — CHANGES_REQUESTED** [review]
Wow! Thank you for implementing the useful rule!

File: packages/eslint-plugin-svelte/src/rules/no-navigation-without-base.ts (hunk 1 above)
Comment (marekdedic) [reply packages/eslint-plugin-svelte/src/rules/no-navigation-without-base.ts]: Great tip, found an issue with the implementation as well

**Comment by ota-meshi** [comment]
Rule implementation looks good, but I made this package ESM.
https://github.com/sveltejs/eslint-plugin-svelte/pull/932

Could you please merge the main branch and fix it by adding `.js` to the import statement of your implementation?

**Review by baseballyama — APPROVED** [review]

**Pull request merged**